### PR TITLE
Initial commit of host metrics network scraper

### PR DESCRIPTION
**Pull request by james-bebbington**
**Link to tracking Issue:**
https://github.com/open-telemetry/opentelemetry-collector/issues/847

**Description:**
Added network scraper to the hostmetricsreceiver which uses gopsutil to collect metrics related to network usage & the number of active tcp connections

**Exported Metrics:**

image

### CODE REVIEW

**Comment by james-bebbington** [comment]
Please review https://github.com/open-telemetry/opentelemetry-collector/pull/926 first - this PR will need a rebase (and very minor fixup to `hostmetrics_receiver_test.go`) after that PR is merged.

Otherwise this is ready to be reviewed now.

File: receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer/pdata"
+	"github.com/open-telemetry/opentelemetry-collector/exporter/exportertest"
+	"github.com/open-telemetry/opentelemetry-collector/receiver/hostmetricsreceiver/internal"
+)
+
+type validationFn func(*testing.T, []pdata.Metrics)
+
+func TestScrapeMetrics(t *testing.T) {
+	createScraperAndValidateScrapedMetrics(t, &Config{}, func(t *testing.T, got []pdata.Metrics) {
```
Comment (bogdandrutu) [inline receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper_test.go:33]: Sorry but again very very hard to read inlined 40 lines functions. Please extract.

**Comment by bogdandrutu** [comment]
Also something bad in the example screenshot the last metric `network_tcp_connections` should be a gauge not a counter. Problem looks to be somewhere in the translation, looks like you do use gauge, did you change that and took a different snapshot?

**Comment by james-bebbington** [comment]
> Also something bad in the example screenshot the last metric `network_tcp_connections` should be a gauge not a counter. Problem looks to be somewhere in the translation, looks like you do use gauge, did you change that and took a different snapshot?

Nice spotting. It looks like that's a general bug in the OC prometheus exporter: https://github.com/orijtech/prometheus-go-metrics-exporter/blob/master/prometheus.go#L315 (also L318) - should be `derivedPrometheusType` instead of `prometheus.CounterValue`

I'll create a PR to fix on that repo tomorrow.

**Pull request merged**